File: task-6/CoffeeShop/CoffeeShop/Models/Order.swift
//
//  Order.swift
//  CoffeeShop
//
//  Created by Victoria Iashchuk on 29/01/2026.
//

import Foundation

struct Order: Codable, Identifiable {
    let id: Int
    let userId: String
    let status: OrderStatus
    let paymentMethod: PaymentMethod
    let createdAt: Date?
    let updatedAt: Date?
    let items: [OrderItem]?
    var totalAmount: Double
}

enum OrderStatus: String, Codable {
    case initialized, processing, completed, cancelled
}

enum PaymentMethod: String, Codable {
    case card, cash
    case applePay = "apple_pay"
    case googlePay = "google_pay"
}

File: task-4/server/src/models/Order.js
import { DataTypes } from 'sequelize'
import sequelize from '../config/database.js'
import User from './User.js'

const Order = sequelize.define('Order', {
    userId: {
        type: DataTypes.INTEGER,
        allowNull: false,
        references: {
            model: User,
            key: 'id'
        }
    },
    status: {
        type: DataTypes.STRING,
        allowNull: false,
        defaultValue: 'initialized',
        validate: {
            isIn: [['initialized', 'processing', 'completed', 'cancelled']]
        }
    },
    totalAmount: {
        type: DataTypes.DECIMAL(10, 2),
        allowNull: false,
        defaultValue: 0.00
    },
    paymentMethod: {
        type: DataTypes.STRING,
        allowNull: false,
        validate: {
            isIn: [['card', 'cash', 'apple_pay', 'google_pay']]
        }
    }
},
    {
        tableName: 'orders'
    })

Order.belongsTo(User, { foreignKey: 'userId', as: 'user' });
User.hasMany(Order, { foreignKey: 'userId', as: 'orders' });

export default Order